### PR TITLE
Update CDI APIs

### PR DESCRIPTION
**Pull request by benjamin-confino**

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build

File: dev/cnf/oss_dependencies.maven [hunk 1]
```diff
@@ -137,7 +137,7 @@ jakarta.batch:jakarta.batch-api:2.0.0
 jakarta.ejb:jakarta.ejb-api:4.0.0
 jakarta.enterprise.concurrent:jakarta.enterprise.concurrent-api:2.0.0
 jakarta.enterprise:jakarta.enterprise.cdi-api:3.0.0
-jakarta.enterprise:jakarta.enterprise.cdi-api:4.0.0-RC3
+jakarta.enterprise:jakarta.enterprise.cdi-api:4.0.0-RC5
 jakarta.enterprise:jakarta.enterprise.lang-model:4.0.0-RC3
```
Comment (Azquelt) [inline dev/cnf/oss_dependencies.maven]: the `lang-model` artifact should be updated at the same time as the `cdi-api`

**Review by Azquelt — CHANGES_REQUESTED** [review]

**Comment by benjamin-confino** [comment]
#build

**Review by Azquelt — APPROVED** [review]

**Comment by Azquelt** [comment]
#libby

**Pull request merged**